### PR TITLE
[Behat][Promotion] Added domain scenarios for promotions deletion

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | yes
| Deprecations?   | no
| Related tickets | partially #4140
| License         | MIT
| Doc PR          | -

### CODE REVIEW

File: features/promotion/prevent_deletion_of_promotion_in_use.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    Background:
+        Given the store operates on a single channel in "France"
+        And the store ships everywhere for free
+        And the store allows paying with "Cash on Delivery"
+        And the store has a product "PHP Mug"
+        And there is a promotion "Holiday promotion"
+        And the promotion gives "10%" percentage discount on shipping to every order
+        And the customer "john.doe@gmail.com" placed an order "#00000022"
+        And the customer chose "Free" shipping method to "France" with "Cash on Delivery" payment
```
Comment (pjedrzejewski) [inline features/promotion/prevent_deletion_of_promotion_in_use.feature]: Free shipping with 10% discount does not make much sense, perhaps we should use flat rate here.

File: features/promotion/deleting_promotion.feature [hunk 2]
```diff
@@ -0,0 +1,15 @@
+@promotion
+Feature: Deleting a promotion
+    In order to delete test or invalid promotions
```
Comment (michalmarcinkowski) [inline features/promotion/deleting_promotion.feature]: We agreed on one version:
`In order to remove test, obsolete or incorrect {resource}`

File: features/promotion/deleting_promotion.feature [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    As an Administrator
+    I want to be able to remove promotions
+
+    Background:
+        Given the store operates on a single channel in "France"
+        And the store has a product "PHP T-Shirt"
```
Comment (michalmarcinkowski) [inline features/promotion/deleting_promotion.feature]: Do we need product here?

File: features/promotion/prevent_deletion_of_promotion_in_use.feature [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        And there is a promotion "Holiday promotion"
+        And it gives "€3.00" fixed discount to every order
+        And the customer "john.doe@gmail.com" placed an order "#00000022"
+        And the customer chose "Free" shipping method to "France" with "Cash on Delivery" payment
+        And the customer bought single "PHP Mug"
+
+    @todo
+    Scenario: Promotion which are in use cannot be deleted
+        When I try to delete promotion "Holiday promotion"
+        Then this promotion should still exist in the registry
```
Comment (michalmarcinkowski) [inline features/promotion/prevent_deletion_of_promotion_in_use.feature]: `And I should be notified that it is in use and cannot be deleted`

File: features/promotion/prevent_deletion_of_promotion_in_use.feature [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        And the store allows paying with "Cash on Delivery"
+        And the store has a product "PHP Mug" priced at "€12.00"
+        And there is a promotion "Holiday promotion"
+        And it gives "€3.00" fixed discount to every order
+        And the customer "john.doe@gmail.com" placed an order "#00000022"
+        And the customer chose "Free" shipping method to "France" with "Cash on Delivery" payment
+        And the customer bought single "PHP Mug"
+
+    @todo
+    Scenario: Promotion which are in use cannot be deleted
```
Comment (michalmarcinkowski) [inline features/promotion/prevent_deletion_of_promotion_in_use.feature]: `Being unable to delete a promotion that was applied to an order`

Comment (NeverResponse) [inline features/promotion/prevent_deletion_of_promotion_in_use.feature]: :+1:

File: features/promotion/prevent_deletion_of_promotion_in_use.feature [hunk 6]
```diff
@@ -0,0 +1,22 @@
+@promotion
+Feature: Prevent deletion of promotions which are in use
+    In order to maintain proper order history
+    As an Administrator
+    I want to be prevented from deleting promotions which are in use
```
Comment (pjedrzejewski) [inline features/promotion/prevent_deletion_of_promotion_in_use.feature]: Promotion in use is not really clear. I'd suggest something like:

```
@promotion
 Feature: Prevent deletion of promotions applied to order
     In order to maintain proper order history
     As an Administrator
     I want to be prevented from deleting promotions which have been already applied to an order
```

WDYT?

Comment (michalmarcinkowski) [inline features/promotion/prevent_deletion_of_promotion_in_use.feature]: :+1:

Comment (NeverResponse) [inline features/promotion/prevent_deletion_of_promotion_in_use.feature]: :+1:

Comment (pamil) [inline features/promotion/prevent_deletion_of_promotion_in_use.feature]: :+1:

**Comment by michalmarcinkowski** [comment]
I think this can be closed since it is not in sync with #4401.

**Pull request closed without merging**